source/tests/system/nixnetsocket_driver_api_tests.cpp

```diff
@@ -27,29 +28,38 @@ constexpr auto INVALID_SOCKET_ERROR = -13008;
 constexpr auto INVALID_SOCKET_MESSAGE = "The specified socket could not be found.";
 constexpr auto INVALID_IP_STACK_ERROR = -13009;
 constexpr auto INVALID_IP_STACK_MESSAGE = "The specified IP Stack could not be found.";
+constexpr auto INVALID_ARGUMENT_ERROR = 13822;
+constexpr auto INVALID_ARGUMENT_MESSAGE = "Invalid argument";
 constexpr auto NXSOCKET_FALSE = 0;
 constexpr auto NXSOCKET_TRUE = 1;
+constexpr auto IPV4_LOCALHOST_ADDR = 0x0100007FU;
+constexpr auto IPV4_LOCALHOST_ADDRESS_STR = "127.0.0.1";
+constexpr auto IPV6_LOCALHOST_ADDRESS_STR = "::1";
+const auto IPV6_LOCALHOST_ADDR = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\x1"s;
+const auto IPV6_ZERO_ADDR = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"s;
 
 constexpr auto SCHEMA = "file:///NIXNET_Documentation/xnetIpStackSchema-03.json";
 const auto SIMPLE_INTERFACE_CONFIG = R"(      
         {
-          "address": "inherit",
+          "address": "generated",
```

Comment:
I tweaked this to be exactly what the nxsocket C API tests use. I don't think most of the changes matter, but I think it's better to have fewer variables to worry about.